torch/csrc/jit/python/python_sugared_value.cpp

```diff
@@ -282,13 +282,44 @@ void recurseThroughNestedModules(
     std::shared_ptr<ModuleValue> self,
     const std::string& prefix,
     const std::string& field) {
-  auto prefix_value =
-      std::make_shared<SimpleValue>(insertConstant(*m.graph(), prefix));
+  checkInterface(loc, m, self, field);
 
-  keys.push_back(prefix_value);
-  values.push_back(self);
+  // The names returned by named_modules and named_parameters are qualified.
+  // submodule_prefix is the qualified name of self with a "." at the end, to
+  // which the names of the submodules or parameters of self will be appended.
+  std::string submodule_prefix = prefix;
+  if (prefix != "") {
+    submodule_prefix = prefix + ".";
+  }
 
-  checkInterface(loc, m, self, field);
+  if (field == "modules" || field == "named_modules") {
+    // If the requested field is modules or named_modules, add the qualified
+    // module name and module to keys and values.
+
+    // prefix is the fully qualified name of self in its module hierarchy.
+    auto prefix_value =
+        std::make_shared<SimpleValue>(insertConstant(*m.graph(), prefix));
+    keys.push_back(prefix_value);
+    values.push_back(self);
+  } else if (field == "named_parameters") {
+    // If the requested field is named_parameters, iterate through the
+    // parameters of self and add the parameter names and parameters (as Graph
+    // Value*'s rather than IValues) to keys and values.
+    auto params_dict = self->getSugaredNamedParameterDict(loc, m);
+    auto param_name_iter = params_dict->keys_;
+    auto param_iter = params_dict->modules_;
+    for (size_t j = 0; j < param_name_iter->tup_.size(); ++j) {
```

Comment:
Nit:
params_dict is not a const, so for-loop needs to calculate `param_name_iter->tup_.size()` in every iteration. You can save this overhead by calculating `param_name_iter->tup_.size()` before loop.

Reply:
Also, this is a case where I really hope we have an `llvm::zip` equivalent in c10.